source/game_sa/GxtChar.h

```
@@ -26,9 +32,28 @@ char* GxtCharToUTF8(char (&out)[N], const GxtChar* src, size_t offset = 0) {
     return GxtCharToUTF8(out, src, N, offset);
 }
 
+inline size_t CalculateUTF8BufferSize(const GxtChar* gxtStr) {
+    size_t len = 0;
+    if (gxtStr) {
+        while (gxtStr[len] != 0) len++;
+        // Modern c++ 20+: dynamic size 4 bytes max per char + null terminator
+        return len * 4 + 1 + 64;
+    }
+    return 0;
```

@yukani this good?

also, please no silent errors here either.
you pass in a null, you get an assert, not `0`.

---

Logic seems safe for UTF-8 chars, (4 chars max.) probably even 2 bytes are fine but I need to check max byte size for chars in `gxtUtf8Map` at `GxtCharToUTF8`. This is fine for now.

I don't get the comment, what modern C++ even matter here?

I agree with @Pirulax here, it should be `NOTSA_UNREACHABLE` even.